src/Gyro.cpp

```diff
@@ -5,100 +5,133 @@
- * */
+// Tilt, roll, and temperature measurementusing the MPU-6050 MEMS gyro.
+// See: https://invensense.tdk.com/products/motion-tracking/6-axis/mpu-6050/
+// Datasheet: https://invensense.tdk.com/wp-content/uploads/2015/02/MPU-6000-Datasheet1.pdf
+// Register descriptions: https://invensense.tdk.com/wp-content/uploads/2015/02/MPU-6000-Register-Map1.pdf
 
-bool Gyro::isPresent(false);
+// Static var to keep track of whether the MPU6050 was found
+bool Gyro::isPresent = false;
 
+// Starts up the MPU-6050 device.
+// Reads the WHO_AM_I register to verify if the device is present.
+// Wakes device from power-down.
+// Sets accelerometers to minimum bandwidth to reduce measurement noise.
 void Gyro::startup()
-/* Starts up the MPU-6050 device.
-   Reads the WHO_AM_I register to verify if the device is present.
-   Wakes device from power-down.
-   Sets accelerometers to minimum bandwidth to reduce measurement noise.
-*/
 {
-    // Initialize interface to the MPU6050
-    LOGV1(DEBUG_INFO, F("GYRO:: Starting"));
+// Initialize interface to the MPU6050
+#if USE_GYRO_WITH_SOFTWAREI2C == 1
+    LOGV1(DEBUG_INFO, F("GYRO:: Starting software I2C for MPU6050 comms."));
+#else
+    LOGV1(DEBUG_INFO, F("GYRO:: Starting hardware I2C for MPU6050 comms."));
+#endif
+
     Wire.begin();
+    Wire.setClock(100000); // Set lowest clock speed (100kHz) to make the communication more reliable
 
+#if USE_GYRO_WITH_SOFTWAREI2C == 0
+    // Wire.setWireTimeout(3000, true); // Set I2C timeout if the Wire.h calls are hanging due to a HW issue
+#endif
     // Execute 1 byte read from MPU6050_REG_WHO_AM_I
     // This is a read-only register which should have the value 0x68
     Wire.beginTransmission(MPU6050_I2C_ADDR);
     Wire.write(MPU6050_REG_WHO_AM_I);
     Wire.endTransmission(true);
     Wire.requestFrom(MPU6050_I2C_ADDR, 1, 1);
-    byte id = (Wire.read() >> 1) & 0x3F;    
+    const byte id = (Wire.read() >> 1) & 0x3F;
     isPresent = (id == 0x34);
-    if (!isPresent) {
-        LOGV1(DEBUG_INFO, F("GYRO:: Not found!"));
+    if (!isPresent)
+    {
+        LOGV2(DEBUG_INFO, F("GYRO:: Not found! Expected 0x34 but got %x"), id);
         return;
     }
 
-    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1
+    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1 to disable sleep and set 8MHz clock
     Wire.beginTransmission(MPU6050_I2C_ADDR);
     Wire.write(MPU6050_REG_PWR_MGMT_1);
-    Wire.write(0);      // Disable sleep, 8 MHz clock
+    Wire.write(0); // Disable sleep, 8 MHz clock
     Wire.endTransmission(true);
 
-    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1
+    // Execute 1 byte read/write to MPU6050_REG_ACCEL_CONFIG to set 2g sensitivity
+    Wire.beginTransmission(MPU6050_I2C_ADDR);
+    Wire.write(MPU6050_REG_ACCEL_CONFIG);
+    Wire.endTransmission();
+    Wire.requestFrom(MPU6050_I2C_ADDR, 1);
+    byte accelConfigReg = Wire.read();              //the value of Register-28 is in x
```

Comment:
Could you do the same thing here as I requested above?
```C++
const byte accelConfigReg = (Wire.read() & 0b11100111);
```

Reply:
Done